examples/hooks-next/pages/index.tsx

```diff
@@ -83,9 +83,8 @@ function FallbackComponent({ attribute }: { attribute: string }) {
 }
 
 export async function getServerSideProps({ req }) {
-  const url = new URL(
-    req.headers.referer || `https://${req.headers.host}${req.url}`
-  ).toString();
+  const protocol = req.headers.referer?.split('://')[0] || 'https';
```

Comment:
```suggestion
  const protocol = new URL(req.headers.referer, 'https://example.com').protocol';
```

does that work or throw an error if referer is not set?